src/shared_modules/indexer_connector/src/indexerConnectorSyncImpl.hpp

```diff
@@ -664,6 +664,52 @@ class IndexerConnectorSyncImpl final
         return resultJson;
     }
 
+    void executeSearchQueryWithPagination(const std::string& index,
```

Comment:
Break conditions are spread across multiple branches with minimal logging. It’s unclear which case terminated the loop when debugging. You could consolidate exit paths and make termination reasons explicit.